app/controllers/application_controller.rb

```diff
@@ -208,6 +208,8 @@ def require_admin
   end
 
   def render_403
+    @missing_permissions = Foreman::AccessControl.permissions_for_controller_action(path_to_authenticate)
```

Comment:
It's not only missing permissions that can trigger render_403, i.e. admin-only actions, missing smart proxy auth and unauthed xhr requestes might trigger this code path.